### ISSUE TITLE
The constructor of AndroidHttpHandler should be preserved together with AndroidMessageHandler

### ISSUE THREAD

**Issue by huoyaoyuan** [xamarin/xamarin-android#6949]
### Android application type

Android for .NET (net6.0-android, etc.)

### Affected platform version

.NET SDK 6.0.202, Runtime 6.0.4

### Description

`Xamarin.Android.Net.AndroidHttpHandler` was used for Xamarin android applications, and is set when using Visual Studio project setting UI. 
However, only the new `AndroidMessageHandler` is preserved:

https://github.com/xamarin/xamarin-android/blob/bacc3f0282b05978f1cde13f0ca49ecc58048efe/src/Mono.Android/Android.Runtime/AndroidEnvironment.cs#L342-L348

When using `HttpClientHandler` with a migrated application, it crashes hardly in Debug configuration, and throws NullReferenceException in Release configuration. It's quite hard to debug for people unfamiliar with how http client handler works.

### Steps to Reproduce

Use `<AndroidHttpClientHandlerType>Xamarin.Android.Net.AndroidHttpHandler</AndroidHttpClientHandlerType>` in any minimal application, and use `HttpClientHandler`.

### Did you find any workaround?

I browsed the code path setting up native handler, and realized only `AndroidMessageHandler` would work.

**Comment by jonathanpeppers** [comment]
@huoyaoyuan `AndroidClientHandler` is marked obsolete:
https://github.com/xamarin/xamarin-android/blob/bacc3f0282b05978f1cde13f0ca49ecc58048efe/src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs#L60-L61

Is there a reason you need to use it? Just so we understand the use-case.

For now, I'd workaround with a custom linker configuration:

https://docs.microsoft.com/en-us/xamarin/cross-platform/deploy-test/linker

```xml
<assembly fullname="Mono.Android">
    <type fullname="Xamarin.Android.Net.AndroidHttpHandler">
        <method signature="System.Void .ctor()" />
    </type>
</assembly>
```

We don't want to put `DynamicDependency`, because that would preserve it in every `net6.0-android` app using `HttpClient`.

**Comment by huoyaoyuan** [comment]
> Is there a reason you need to use it? Just so we understand the use-case.

Converted legacy project specifies this handler. Supporting it would make migration easier.
It's also nice to include a sdk warning when the handler is specified, and update Visual Studio project UI to select the new one.

**Comment by jonathanpeppers** [comment]
@huoyaoyuan were you using the .NET upgrade assistant?

I'd recommend removing almost every MSBuild property in the project file, and just use the defaults.

We have some of this documented here:

https://github.com/xamarin/xamarin-android/blob/main/Documentation/guides/OneDotNet.md#changes-to-msbuild-tasks

Or you can see the template as what we pretty much recommend:

https://github.com/xamarin/xamarin-android/blob/bacc3f0282b05978f1cde13f0ca49ecc58048efe/src/Microsoft.Android.Templates/android/AndroidApp1.csproj#L1-L13

**Comment by akoeplinger** [comment]
Just to make sure: you used `Xamarin.Android.Net.AndroidClientHandler` not `Xamarin.Android.Net.AndroidHttpHandler` in the `AndroidHttpClientHandlerType` property right?

**Comment by huoyaoyuan** [comment]
> Just to make sure: you used `Xamarin.Android.Net.AndroidClientHandler` not `Xamarin.Android.Net.AndroidHttpHandler` in the `AndroidHttpClientHandlerType` property right?

Confirmed in commit history that it was `<AndroidHttpClientHandlerType>Xamarin.Android.Net.AndroidClientHandler</AndroidHttpClientHandlerType>`.


> were you using the .NET upgrade assistant?

No. I manually converted it based on my knowledge of MSBuild. The project file was somehow customized with `.props`.

The VS project setting UI (in VS 17.1) sets `AndroidClientHandler` when selecting in http handler. This should be probably updated to the recommended value too.
